torch/csrc/distributed/autograd/context/dist_autograd_context.h

```diff
@@ -29,19 +34,44 @@ class TORCH_API DistAutogradContext {
       std::shared_ptr<RecvRpcBackward>& func,
       int64_t autograd_message_id);
 
+  // Given an autograd_message_id, retrieve the appropriate send function.
+  std::shared_ptr<SendRpcBackward> retrieveSendFunction(
+      int64_t autograd_message_id);
+
+  // Receive all send functions for this context.
   std::unordered_map<int64_t, std::shared_ptr<SendRpcBackward>> sendFunctions()
       const;
 
+  // Receive all recvfunctions for this context.
   std::unordered_map<int64_t, std::shared_ptr<RecvRpcBackward>> recvFunctions()
       const;
 
+  // Record that we would like to accumulate the provided gradient on the given
+  // variable.
+  void accumulateGrad(torch::autograd::Variable variable, torch::Tensor grad);
+
+  // Retrieve the GraphTask.
+  torch::autograd::GraphTask& retrieveGraphTask();
+
+  // Set the appropriate graph task.
+  void setGraphTask(std::unique_ptr<torch::autograd::GraphTask> graphTask);
```

Comment:
I am immensely doubtful this should be in the public API. If I understand correctly, you only ever call this once, as part of initialization. Adding `setGraphTask` to the public API implies that you can change it midstream; that seems hardly intentional (and also very unlikely to work...)

Reply:
C++ gives you plenty of tools for this situation right? For example, a friend function seems suitable. Also renaming the function so it's clear that it is a one-shot thing would also help. Or even a comment!